### ISSUE TITLE
Crop or Pad transform.

### ISSUE THREAD

**Issue by ryanpeach** [pytorch/vision#562]
So, tensorflow has a transform function called [resize_image_with_crop_or_pad](https://www.tensorflow.org/api_docs/python/tf/image/resize_image_with_crop_or_pad). I recently needed this function for my own work. Moreover, it'd be nice to have all of the following:

* RandomPad - Makes an image a certain size by padding but performs random offset on the padding.
* CenterCropOrPad - Replicates Tensorflows function
* RandomCropOrPad - A conditional mix of RandomCrop and RandomPad, guaranteeing size but w random offset.

If I put in a pull request would you be open to these methods?

What are the standards for writing transforms?

**Comment by fmassa** [comment]
Isn't [`RandomCrop` what you want](https://github.com/pytorch/vision/blob/master/torchvision/transforms/transforms.py#L373-L401)?

**Comment by ryanpeach** [comment]
I tried RandomCrop and it threw an error when the desired size was bigger than the source. Is it supposed to be able to pad also?

**Comment by fmassa** [comment]
Yes, latest version of `torchvision` supports padding as well, check the link I sent you. There is an option for optionally adding padding if the input is too small

**Comment by ryanpeach** [comment]
Oh I see in the documentation it is, never mind, maybe the bug was something else then. Maybe there's a way to change the description of this method in the list of transforms so that it mentions padding as well? (Maybe that's the case and I missed it as well).

Sorry, just now learning the library. I'll try to be more careful before making issues next time.

**Comment by fmassa** [comment]
No worries.
We can definitely improve the documentation. Could you please send a PR?

**Comment by ryanpeach** [comment]
Yes I'll look into it.

**Comment by ryanpeach** [comment]
So I also learned that RandomCrop when you set `pad_if_needed` to True does not RandomPad. AKA, whereas random crop applies a crop at a random offset to the image, `pad_if_needed` performs a centered padding, which does not randomly place the image in the padded image. The goal for padding should be that the image is in a random spot. Imagine the following:

Imagine the image, whatever size, on an infinite canvas.
Apply a random offset, and a random crop, to that canvas, such that:
if the crop is bigger than the original image, at least some of the original image remains in the output.
if the crop is smaller than the original image, all of the output is part of the original image.

**Comment by ryanpeach** [comment]
The easy fix for this over-pad the image in RandomCrop.

**Comment by fmassa** [comment]
The random crop behavior was fixed by #564. @ryanpeach could you also send a PR improving the documentation?

**Comment by ryanpeach** [comment]
Yes sorry, I have limited time for that but next time I work on my project I'll get to it!

**Comment by ekagra-ranjan** [comment]
Can be closed via #791.

**Comment by soumith** [comment]
closed via #791
